code/modules/antagonists/eldritch_cult/knowledge.dm/rust_lore.dm

```diff
@@ -0,0 +1,248 @@
+	route = "Rust"
+
+/datum/eldritch_knowledge/rust_regen
+	name = "Leeching Walk"
+	desc = "Passively heals you when you are on rusted tiles."
+	gain_text = "The strength was unparallel, it was unnatural. Blacksmith was smiling."
+	cost = 1
+	next_knowledge = list(/datum/eldritch_knowledge/rust_mark,/datum/eldritch_knowledge/armor,/datum/eldritch_knowledge/essence)
+	route = "Rust"
+
+/datum/eldritch_knowledge/rust_regen/on_life(mob/user)
+	. = ..()
+	var/turf/T = get_turf(user)
+	if(!istype(T,/turf/open/floor/plating/rust) || !isliving(user))
+		return
+	var/mob/living/L = user
+	L.adjustBruteLoss(-2)
+	L.adjustFireLoss(-2)
+	L.adjustToxLoss(-2)
+	L.adjustOxyLoss(-0.5)
+	return
+
+/datum/eldritch_knowledge/rust_mark
+	name = "Mark of Rust"
+	desc = "Your eldritch blade now applies a mark. To Detonate the mark use your mansus grasp on it."
+	gain_text = "Lords of the depths help those in dire need at a cost."
+	cost = 2
+	next_knowledge = list(/datum/eldritch_knowledge/spell/area_conversion)
+	banned_knowledge = list(/datum/eldritch_knowledge/ash_mark,/datum/eldritch_knowledge/flesh_mark)
+	route = "Rust"
+
+/datum/eldritch_knowledge/rust_mark/eldritch_blade_act(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(istype(target,/mob/living))
+		var/mob/living/L = target
+		L.apply_status_effect(/datum/status_effect/eldritch/rust)
+
+/datum/eldritch_knowledge/rust_blade_upgrade
+	name = "Toxic blade"
+	gain_text = "Let the blade guide you through the flesh."
+	desc = "Your blade of choice will now add toxin to your enemies system."
+	cost = 2
+	next_knowledge = list(/datum/eldritch_knowledge/spell/rust_wave)
+	banned_knowledge = list(/datum/eldritch_knowledge/ash_blade_upgrade,/datum/eldritch_knowledge/flesh_blade_upgrade)
+	route = "Rust"
+
+/datum/eldritch_knowledge/rust_blade_upgrade/eldritch_blade_act(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		C.reagents.add_reagent(/datum/reagent/toxin,5)
+
+/datum/eldritch_knowledge/spell/rust_wave
+	name = "Wave of Rust"
+	desc = "You can now send a projectile that converts an area into rust."
+	gain_text = "Messenger's of hope fear the rustbringer!"
+	cost = 1
+	spell_to_add = /obj/effect/proc_holder/spell/targeted/projectile/dumbfire/rust_wave
+	next_knowledge = list(/datum/eldritch_knowledge/rust_final,/datum/eldritch_knowledge/spell/blood_siphon,/datum/eldritch_knowledge/summon/rusty)
+	route = "Rust"
+
+/datum/eldritch_knowledge/armor
+	name = "Armorer's ritual"
+	desc = "You can now create eldritch armor using a table and a kitchen knife."
+	gain_text = "For I am the heir to the throne of doom."
+	cost = 1
+	next_knowledge = list(/datum/eldritch_knowledge/rust_regen,/datum/eldritch_knowledge/flesh_ghoul)
+	required_atoms = list(/obj/structure/table,/obj/item/kitchen/knife)
```

Comment:
does this consume the table?

Reply:
yes

Reply:
all atoms are destroyed unless cleanup_atoms() is overridden